pallets/attesters/src/lib.rs

```diff
@@ -0,0 +1,864 @@
+                Error::<T>::NominatorNotEnoughBalance
+            );
+
+            nomination.1 += amount;
+
+            let nomination_amount = nomination.1;
+
+            // Update the nomination storage item
+            Nominations::<T>::insert(&attester, nomination);
+
+            // Update the sorted list of nominated attesters
+            SortedNominatedAttesters::<T>::try_mutate(|attesters| {
+                if let Some(index) = attesters.iter().position(|(a, _n)| a == &attester) {
+                    // Update the existing nomination amount
+                    attesters[index] = (attester.clone(), nomination_amount);
+                } else {
+                    // Add the new attester to the list
+                    attesters.push((attester.clone(), nomination_amount));
+                }
+
+                // Sort the attesters by their nomination amount
+                attesters.sort_by(|(_a1, n1), (_a2, n2)| n2.cmp(n1));
+
+                // Keep only the top 32 attesters in the list
+                attesters.truncate(32);
+
+                Ok::<(), Error<T>>(())
+            })?;
+
+            // Lock the nomination amount in the nominator's account
+            T::Currency::reserve(&nominator, amount)?;
+
+            Ok(())
+        }
+
+        fn shuffle_committee() -> bool {
+            let active_set = ActiveSet::<T>::get();
+            let active_set_size = active_set.len();
+            let mut committee_size = T::CommitteeSize::get() as usize;
+
+            let full_shuffle = if committee_size > active_set_size {
+                committee_size = active_set_size;
+                false
+            } else {
+                true
+            };
+
+            let current_committee = CurrentCommittee::<T>::get();
+            PreviousCommittee::<T>::put(current_committee);
+
+            let _seed = T::RandomnessSource::random_seed();
+
+            let mut shuffled_active_set = active_set;
+            for i in (1..shuffled_active_set.len()).rev() {
+                let random_value = T::RandomnessSource::random(&i.to_be_bytes());
+                let random_index = random_value
+                    .0
+                    .as_ref()
+                    .iter()
+                    .fold(0usize, |acc, &val| (acc + val as usize) % (i + 1));
+
+                if i != random_index {
+                    shuffled_active_set.swap(i, random_index);
+                }
+            }
+
+            let new_committee = shuffled_active_set
+                .into_iter()
+                .take(committee_size)
+                .collect::<Vec<T::AccountId>>();
+
+            CurrentCommittee::<T>::put(new_committee);
+
+            full_shuffle
+        }
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(n: T::BlockNumber) -> Weight {
+            // Check if a shuffling round has passed
+            if (n % T::ShufflingFrequency::get()).is_zero() {
```

Comment:
We should consider always assigning the new attestor set I shuffling period in advance, so that we always have current and next committee. And then we need to generate a proof, where the current committee signes of the next